packages/client/src/provider/in-memory-provider/in-memory-provider.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  FlagValueType,
+  GeneralError,
+  JsonValue,
+  Logger,
+  OpenFeatureError,
+  ProviderEvents,
+  ResolutionDetails,
+  StandardResolutionReasons,
+  TypeMismatchError,
+  ProviderStatus,
+} from '@openfeature/core';
+import { Provider } from '../provider';
+import { OpenFeatureEventEmitter } from '../../events';
+import { FlagConfiguration, Flag } from './flag-configuration';
+import { VariantFoundError } from './variant-not-found-error';
+
+/**
+ * A simple OpenFeature provider intended for demos and as a test stub.
+ */
+export class InMemoryProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
+  public readonly runsOn = 'client';
+  status: ProviderStatus = ProviderStatus.NOT_READY;
+  readonly metadata = {
+    name: 'in-memory',
+  } as const;
+  private _flagConfiguration: FlagConfiguration;
+  private _context: EvaluationContext | undefined;
+
+  constructor(flagConfiguration: FlagConfiguration = {}) {
+    this._flagConfiguration = { ...flagConfiguration };
+  }
+
+  async initialize(context?: EvaluationContext | undefined): Promise<void> {
+    try {
+
+      for (const key in this._flagConfiguration) {
+        this.resolveFlagWithReason(key, context);
+      }
+
+      this._context = context;
+      this.status = ProviderStatus.READY;
+    } catch (error) {
+      this.status = ProviderStatus.ERROR;
+      throw error;
```

Comment:
Originally you were manually emitting ready/error events here, which the SDK doesn't based on the resolve/reject of this promise. This caused duplicate events.

Reply:
It may be worth adding this as a comment to the provider since people may use the in-memory provider as a point of reference.